### DISCUSSION TITLE
About the compatibility of Colossal and torch.distributed.

### DISCUSSION THREAD

**Discussion by ZhuXMMM** [hpcaitech/ColossalAI#3460] [Community | Q&A]
In my code, the original distributed sequences were using functions such as torch.distributed.all_reduce(), torch.distributed.get_backend(), torch.distributed.all_gather(), torch.distributed.is_available(), etc. I want to ask whether these functions are compatible with Colossal. If I want to replace them, which functions in Colossal should I use, or how can I find out?

**Comment by kurisusnowdeng** [comment]
We maintain the similar information of all distributed groups in a [global context](https://github.com/hpcaitech/ColossalAI/blob/6650daeb0ae43ab93e2341cce75975f3e553c02a/colossalai/context/parallel_context.py#L25). You shall use `gpc` in your codes like `from colossalai.core import global_context as gpc`. Note that a [parallel mode](https://github.com/hpcaitech/ColossalAI/blob/main/colossalai/context/parallel_mode.py) is usually needed to specify a certain group
